File: Spring/spring_session/src/main/java/com/springsession/spring_session/service/BoardService.java
package com.springsession.spring_session.service;

import com.springsession.spring_session.entity.Board;
import com.springsession.spring_session.repository.BoardRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class BoardService {
    private BoardRepository boardRepository;

    @Autowired
    public BoardService(BoardRepository boardRepository) {
        this.boardRepository = boardRepository;
    }

    public void write(Board board) {
        boardRepository.save(board);
    }
}

File: 03_Spring/session/src/main/java/gdsc/session/service/BoardService.kt
package gdsc.session.service

import gdsc.session.entity.Board
import gdsc.session.repository.BoardRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Service

@Service
class BoardService(@Autowired val boardRepository: BoardRepository) {
    fun write(board: Board): Board = boardRepository.save(board)
    fun boardList(pageable: Pageable): Page<Board> = boardRepository.findAll(pageable)
    fun boardView(id: Int) = boardRepository.findById(id).get()

    fun boardSearchList(searchKeyword: String, pageable: Pageable): Page<Board> =
        boardRepository.findByTitleContaining(searchKeyword, pageable)
    fun boardDelete(id: Int) = boardRepository.deleteById(id)
}